internal/auth/nil_auth.go

```diff
@@ -0,0 +1,19 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+)
+
+type NilAuth struct{}
```

Comment:
I assume when choosing `NilAuth` it means "run without external auth backend", but not "without authentication", or? IOW, we'd still want to generate an admin account (read from a Secret) that a user needs to log in with. Any ideas how you'd plug this into this framework? (Can be a follow-up PR.)

Reply:
"we'd still want to generate an admin account (read from a Secret) that a user needs to log in"

Doing that will require basic authentication middleware, which can verify the user/password and allow authentication. 
We need to have a separate route for basic auth and a route for token-based authentication

@fzdarsky @mangelajo do you think a separate route is possible?

Reply:
I changed this to a basic auth.
The password is generated on runtime and stored in `config.yaml` (user can also override it to define a custom one).
Would this be enough for the OOTB auth ?

Reply:
...aand back to NilAuth which we would use for the development envs.

Reply:
Hahaha, thank you. Just for the records: We decided not to support basic auth, but deploy Keycloak as default auth server.